easybuild/framework/easyconfig/tweak.py

```diff
@@ -93,18 +93,46 @@ def tweak(easyconfigs, build_specs, modtool, targetdirs=None):
         _log.debug("Applying build specifications recursively (no software name/version found): %s" % build_specs)
         orig_ecs = resolve_dependencies(easyconfigs, modtool, retain_all_deps=True)
 
-    # keep track of originally listed easyconfigs (via their path)
-    listed_ec_paths = [ec['spec'] for ec in easyconfigs]
-
-    # determine toolchain based on last easyconfigs
-    if orig_ecs:
-        toolchain = orig_ecs[-1]['ec']['toolchain']
-        _log.debug("Filtering using toolchain %s" % toolchain)
+        # determine original toolchain based on last easyconfig
+        tcname = orig_ecs[-1]['ec']['toolchain']['name']
+        _log.debug("Filtering using toolchain %s", tcname)
 
         # filter easyconfigs unless a dummy toolchain is used: drop toolchain and toolchain dependencies
-        if toolchain['name'] != DUMMY_TOOLCHAIN_NAME:
-            while orig_ecs[0]['ec']['toolchain'] != toolchain:
-                orig_ecs = orig_ecs[1:]
+        if tcname != DUMMY_TOOLCHAIN_NAME:
+
+            # locate toolchain itself, there should be exactly one match
+            tc_ecs = [ec for ec in orig_ecs if ec['ec']['name'] == tcname]
+            if len(tc_ecs) == 1:
+                tc_ec = tc_ecs[0]
+            else:
+                raise EasyBuildError("Failed to isolate toolchain '%s' in dependency graph: %s", tcname, orig_ecs)
+
+            # FIXME: need to actually construct dep graph without toolchain,
```

Comment:
I don't think this is necessary. Only explicitly listed easyconfigs are tweaked and put first in the search, everything else is tweaked and placed last in the search (i.e., they're only used if they don't already exist). When it comes to the toolchain, that will have been replaced and so once the deps of the tweaked easyconfigs (that appeared on the command line) are resolved the tweaked toolchain easyconfigs will never get used.